x/stakeibc/keeper/reward_converter.go

```diff
@@ -57,75 +57,95 @@ type FeeInfo struct {
 	if !chainHasRebate {
-		feeInfo = FeeInfo{
-			RebateAmount:    sdkmath.ZeroInt(),
-			StrideFeeAmount: totalFeesAmount,
-			ReinvestAmount:  reinvestAmount,
-		}
-		return feeInfo, nil
+		return sdkmath.ZeroInt(), rewardAmount, nil
 	}
 
+	// Get the fee percentage from params
+	strideFee := sdk.NewIntFromUint64(k.GetParams(ctx).StrideCommission)
+	strideFeePercentage := sdk.NewDecFromInt(strideFee).Quo(sdk.NewDec(100))
+
 	// It shouldn't be possible to have 0 delegations (since there are rewards and there was a community pool stake)
 	// This will also prevent a division by 0 error
 	if hostZone.TotalDelegations.IsZero() {
-		return feeInfo, errorsmod.Wrapf(types.ErrDivisionByZero,
+		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrDivisionByZero,
 			"unable to calculate rebate amount for %s since total delegations are 0", chainId)
 	}
 
 	// It also shouldn't be possible for the liquid stake amount to be greater than the full TVL
 	if rebateInfo.LiquidStakeAmount.GT(hostZone.TotalDelegations) {
-		return feeInfo, errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
+		return sdkmath.ZeroInt(), sdkmath.ZeroInt(), errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed,
 			"community pool liquid staked amount greater than total delegations")
 	}
 
 	// The rebate amount is determined by the contribution of the community pool stake towards the total TVL,
 	// multiplied by the rebate fee percentage
 	contributePercentage := sdk.NewDecFromInt(rebateInfo.LiquidStakeAmount).Quo(sdk.NewDecFromInt(hostZone.TotalDelegations))
-	rebateAmount := sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
-	strideFeeAmount := totalFeesAmount.Sub(rebateAmount)
+	totalFeesAmount := sdk.NewDecFromInt(rewardAmount).Mul(strideFeePercentage).TruncateInt()
+	rebateAmount = sdk.NewDecFromInt(totalFeesAmount).Mul(contributePercentage).Mul(rebateInfo.RebatePercentage).TruncateInt()
+	remainingAmount = rewardAmount.Sub(rebateAmount)
 
-	// Sanity check calculation
-	if !rebateAmount.Add(strideFeeAmount).Add(reinvestAmount).Equal(rewardAmount) {
-		return feeInfo, errorsmod.Wrapf(types.ErrFeeSplitInvariantFailed, "fee info compontents do not sum to rewards")
-	}
+	return rebateAmount, remainingAmount, nil
+}
 
-	feeInfo = FeeInfo{
-		RebateAmount:    rebateAmount,
-		StrideFeeAmount: strideFeeAmount,
-		ReinvestAmount:  reinvestAmount,
+// Given the reward balance from an ICQ, calculates the portions intended as a stride fee vs for reinvestment
+//
+// If the chain doesn't have a rebate in place, the split is decided entirely from the stride commission percent
+// However, if the chain does have a rebate, we need to factor that into the calculation
+//
+// For instance, if 100 rewards were collected and 5 were sent as a rebate, then the stride fee should be based
+// on the original 100 rewards instead of the remaining 95
+func (k Keeper) CalculateRewardsFeeSplit(
```

Comment:
If I understand correctly, `CalculateRewardsFeeSplit` is used when calculating the split on the DYDX rewards in WithdrawalBalanceCallback, and `CalculateRewardsRebateSplit` is used to calculate the split on USDC rewards. Finding the names a bit confusing - had to trace the code for calls to this and `CalculateRewardsRebateSplit` to understand which was which. Possible I'm missing some detail that makes them clear - one idea would be to name them according to the functions they're called in e.g. `CalculateRewardsSplitHostDenom` or similar.

Reply:
done

Reply:
`CalculateRewardsSplitHostDenom` lgtm, though `CalculateRewardsSplitRewardDenom` still confusing to me. Opened a new comment about that so will resolve this.